From a2832bf95bf0fa4e866ee4a04cabe944d5374ffb Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Mon, 1 Dec 2014 11:48:39 +0100
Subject: [PATCH] nathelper: use dedicated function for testing via address
 against source ip

- reported by Igor Potjevlesch
- previous used functions was checking the presence of rport and
  received parameters, ending in a wrong result for nat testing

(cherry picked from commit 0b10377df11977d7ddd7e6bf30fe060d1fe200aa)
---
 modules/nathelper/nathelper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/nathelper/nathelper.c b/modules/nathelper/nathelper.c
index 7626ed6..0cda937 100644
--- a/modules/nathelper/nathelper.c
+++ b/modules/nathelper/nathelper.c
@@ -1556,7 +1556,7 @@ nat_uac_test_f(struct sip_msg* msg, char* str1, char* str2)
 	 * test if source address of signaling is different from
 	 * address advertised in Via
 	 */
-	if ((tests & NAT_UAC_TEST_RCVD) && received_test(msg))
+	if ((tests & NAT_UAC_TEST_RCVD) && received_via_test(msg))
 		return 1;
 	/*
 	 * test for occurrences of RFC1918 addresses in Contact
-- 
2.1.3

